src/sync.rs

```diff
@@ -713,6 +729,47 @@ fn test_non_lockfree_unchecked() {
     LockFreeFrozenVec::<()>::new();
 }
 
+impl<T: Copy + Clone> Clone for LockFreeFrozenVec<T> {
+    fn clone(&self) -> Self {
+        let len = self.len.load(Ordering::Acquire);
+        // handle the empty case
+        if len == 0 {
+            return Self::default();
+        }
+
+        // copy the data
+        let data = [Self::NULL; NUM_BUFFERS];
+        // for each buffer, copy the data
+        for i in 0..NUM_BUFFERS {
+            // get the buffer size and index
+            let buffer_size = buffer_size(i);
+            let buffer_idx = buffer_index(buffer_size - 1);
+            // get the buffer pointer
+            let buffer_ptr = self.data[buffer_idx].load(Ordering::Acquire);
```

Comment:
don't you want

```suggestion
            let buffer_ptr = self.data[i].load(Ordering::Acquire);
```

here?

Reply:
Fixed in [`efea7c4` (#57)](https://github.com/Manishearth/elsa/pull/57/commits/efea7c410df11070a7c145793793069f35ac25c7)